### ISSUE TITLE
Pylint's "usage" output does not say you have to specify the files to scan.

### ISSUE THREAD

**Issue by ax-jrobb** [pylint-dev/pylint#8922]
### Bug description

If you just run `pylint`, you get the usage/help output, with exit code 32 (options error).

The actual problem is that you haven't specified any files, but literally nothing about this output communicates that fact. `--long-help` doesn't either.

### Command used

```shell
pylint
```

### Pylint output

```shell
The first line of the output is:

usage: pylint [options]


Followed by the standard list of options and switches -- but not one single mention of the need to specify files.
```

### Expected behavior

IMO, the usage text should be:
```plain
usage: pylint [options] [files]
```

This should be accompanied by an explanation that `files` supports `**`, and that it is required unless you're specifying an option that causes pylint _not_ to need those (e.g. because you used `--from-stdin`, or a command that doesn't involve a scan, such as `--help`, `--full-documentation`, etc.).

And, if you invoke `pylint` without specifying any files to scan, the usage text should be preceded by an error message saying you didn't specify any files to scan.

### Alternative solution

Alternatively, pylint could scan the current directory recursively by default, i.e. the equivalent of:
```sh
pylint "$(pwd)"'/**/*.py'
```

This should still be paired with an update to the usage information to make clear that the user can specify a list of files.

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.4 (main, Jul 28 2023, 14:44:21) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### OS / Environment

macOS 13.4.1

**Comment by Pierre-Sassoulas** [comment]
Duplicate of #5701 this is going to be fixed in 3.0
